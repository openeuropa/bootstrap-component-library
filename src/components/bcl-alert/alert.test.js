import {
  merge,
  renderTwigFileAsNode,
  getVariants,
} from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-data-alert/data";

const template = "@oe-bcl/bcl-alert/alert.html.twig";
const render = (params) => renderTwigFileAsNode(template, params, true);
const variants = getVariants();

describe("OE - Alert", () => {
  test(`renders correctly without close button`, () => {
    expect.assertions(1);

    return expect(
      render({
        ...demoData,
        dismissible: false,
      })
    ).resolves.toMatchSnapshot();
  });

  test(`renders correctly without animation for the close button`, () => {
    expect.assertions(1);

    return expect(
      render({
        ...demoData,
        animated_dismiss: false,
      })
    ).resolves.toMatchSnapshot();
  });

  variants.forEach((variant) => {
    test(`${variant} renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render({
          ...demoData,
          variant: variant,
        })
      ).resolves.toMatchSnapshot();
    });
  });
});
