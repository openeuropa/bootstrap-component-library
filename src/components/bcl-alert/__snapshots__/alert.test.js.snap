// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Alert danger renders correctly 1`] = `
<jest>
  <div
    class="alert alert-danger d-flex align-items-center alert-dismissible fade show"
    role="alert"
  >
    <svg
      class="flex-shrink-0 me-3 mt-1 align-self-start bi icon--s"
    >
      <use
        xlink:href="/icons.svg#dash-circle-fill"
      />
    </svg>
    <div
      class="alert-content flex-grow-1"
    >
      <h4
        class="alert-heading"
      >
        Alert heading
      </h4>
      A simple alert. check it out!
    </div>
    <button
      aria-label="Close"
      class="btn-close"
      data-bs-dismiss="alert"
      type="button"
    />
  </div>
</jest>
`;

exports[`OE - Alert dark renders correctly 1`] = `
<jest>
  <div
    class="alert alert-dark d-flex align-items-center alert-dismissible fade show"
    role="alert"
  >
    <svg
      class="flex-shrink-0 me-3 mt-1 align-self-start bi icon--s"
    >
      <use
        xlink:href="/icons.svg#info-circle-fill"
      />
    </svg>
    <div
      class="alert-content flex-grow-1"
    >
      <h4
        class="alert-heading"
      >
        Alert heading
      </h4>
      A simple alert. check it out!
    </div>
    <button
      aria-label="Close"
      class="btn-close"
      data-bs-dismiss="alert"
      type="button"
    />
  </div>
</jest>
`;

exports[`OE - Alert info renders correctly 1`] = `
<jest>
  <div
    class="alert alert-info d-flex align-items-center alert-dismissible fade show"
    role="alert"
  >
    <svg
      class="flex-shrink-0 me-3 mt-1 align-self-start bi icon--s"
    >
      <use
        xlink:href="/icons.svg#info-circle-fill"
      />
    </svg>
    <div
      class="alert-content flex-grow-1"
    >
      <h4
        class="alert-heading"
      >
        Alert heading
      </h4>
      A simple alert. check it out!
    </div>
    <button
      aria-label="Close"
      class="btn-close"
      data-bs-dismiss="alert"
      type="button"
    />
  </div>
</jest>
`;

exports[`OE - Alert light renders correctly 1`] = `
<jest>
  <div
    class="alert alert-light d-flex align-items-center alert-dismissible fade show"
    role="alert"
  >
    <svg
      class="flex-shrink-0 me-3 mt-1 align-self-start bi icon--s"
    >
      <use
        xlink:href="/icons.svg#info-circle-fill"
      />
    </svg>
    <div
      class="alert-content flex-grow-1"
    >
      <h4
        class="alert-heading"
      >
        Alert heading
      </h4>
      A simple alert. check it out!
    </div>
    <button
      aria-label="Close"
      class="btn-close"
      data-bs-dismiss="alert"
      type="button"
    />
  </div>
</jest>
`;

exports[`OE - Alert primary renders correctly 1`] = `
<jest>
  <div
    class="alert alert-primary d-flex align-items-center alert-dismissible fade show"
    role="alert"
  >
    <svg
      class="flex-shrink-0 me-3 mt-1 align-self-start bi icon--s"
    >
      <use
        xlink:href="/icons.svg#info-circle-fill"
      />
    </svg>
    <div
      class="alert-content flex-grow-1"
    >
      <h4
        class="alert-heading"
      >
        Alert heading
      </h4>
      A simple alert. check it out!
    </div>
    <button
      aria-label="Close"
      class="btn-close"
      data-bs-dismiss="alert"
      type="button"
    />
  </div>
</jest>
`;

exports[`OE - Alert renders correctly without animation for the close button 1`] = `
<jest>
  <div
    class="alert alert-primary d-flex align-items-center alert-dismissible"
    role="alert"
  >
    <svg
      class="flex-shrink-0 me-3 mt-1 align-self-start bi icon--s"
    >
      <use
        xlink:href="/icons.svg#info-circle-fill"
      />
    </svg>
    <div
      class="alert-content flex-grow-1"
    >
      <h4
        class="alert-heading"
      >
        Alert heading
      </h4>
      A simple alert. check it out!
    </div>
    <button
      aria-label="Close"
      class="btn-close"
      data-bs-dismiss="alert"
      type="button"
    />
  </div>
</jest>
`;

exports[`OE - Alert renders correctly without close button 1`] = `
<jest>
  <div
    class="alert alert-primary d-flex align-items-center fade show"
    role="alert"
  >
    <svg
      class="flex-shrink-0 me-3 mt-1 align-self-start bi icon--s"
    >
      <use
        xlink:href="/icons.svg#info-circle-fill"
      />
    </svg>
    <div
      class="alert-content flex-grow-1"
    >
      <h4
        class="alert-heading"
      >
        Alert heading
      </h4>
      A simple alert. check it out!
    </div>
  </div>
</jest>
`;

exports[`OE - Alert secondary renders correctly 1`] = `
<jest>
  <div
    class="alert alert-secondary d-flex align-items-center alert-dismissible fade show"
    role="alert"
  >
    <svg
      class="flex-shrink-0 me-3 mt-1 align-self-start bi icon--s"
    >
      <use
        xlink:href="/icons.svg#info-circle-fill"
      />
    </svg>
    <div
      class="alert-content flex-grow-1"
    >
      <h4
        class="alert-heading"
      >
        Alert heading
      </h4>
      A simple alert. check it out!
    </div>
    <button
      aria-label="Close"
      class="btn-close"
      data-bs-dismiss="alert"
      type="button"
    />
  </div>
</jest>
`;

exports[`OE - Alert success renders correctly 1`] = `
<jest>
  <div
    class="alert alert-success d-flex align-items-center alert-dismissible fade show"
    role="alert"
  >
    <svg
      class="flex-shrink-0 me-3 mt-1 align-self-start bi icon--s"
    >
      <use
        xlink:href="/icons.svg#check-circle-fill"
      />
    </svg>
    <div
      class="alert-content flex-grow-1"
    >
      <h4
        class="alert-heading"
      >
        Alert heading
      </h4>
      A simple alert. check it out!
    </div>
    <button
      aria-label="Close"
      class="btn-close"
      data-bs-dismiss="alert"
      type="button"
    />
  </div>
</jest>
`;

exports[`OE - Alert warning renders correctly 1`] = `
<jest>
  <div
    class="alert alert-warning d-flex align-items-center alert-dismissible fade show"
    role="alert"
  >
    <svg
      class="flex-shrink-0 me-3 mt-1 align-self-start bi icon--s"
    >
      <use
        xlink:href="/icons.svg#exclamation-triangle-fill"
      />
    </svg>
    <div
      class="alert-content flex-grow-1"
    >
      <h4
        class="alert-heading"
      >
        Alert heading
      </h4>
      A simple alert. check it out!
    </div>
    <button
      aria-label="Close"
      class="btn-close"
      data-bs-dismiss="alert"
      type="button"
    />
  </div>
</jest>
`;
